src/Symfony/Component/DependencyInjection/Argument/ServiceLocatorArgument.php

```diff
@@ -39,7 +39,7 @@ public function getValues()
     public function setValues(array $values)
     {
         foreach ($values as $v) {
-            if (!$v instanceof Reference) {
+            if (!$v instanceof Reference && null !== $v) {
```

Comment:
Self answer: bad idea: maybe the consumer expects a value from get for the provided key. :)
Test case missing?